app/models/like.rb

```diff
@@ -0,0 +1,12 @@
+class Like < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  belongs_to :post
+
+  def self.likes_counter(post)
+    where(post:).count
+  end
+
+  def self.update_likes_counter(post)
+    post.update(likes_counter: likes_counter(post))
+  end
```

Comment:
- [ ] Kindly change this to an instance method since it should be called on the like instance. It is also more intuitive. Here is a suggestion.

  ```
  def update_likes_counter
    post.increment!(:likes_counter)
  end
  ```

- [ ] **[OPTIONAL]** We can also include a callback for this method so that it is triggered whenever a like is saved, without us having to call it ourselves.

  ```
  belongs_to :author, class_name: 'User'
  belongs_to :post
  after_save :update_likes_counter

  ...

  private

  def update_likes_counter
    post.increment!(:likes_counter)
  end
  ```